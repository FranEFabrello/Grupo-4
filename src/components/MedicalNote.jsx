import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';
import ProfileField from './ProfileField';

export default function MedicalNote({ doctor, date, reason, diagnosis, notes, prescription, onDownload }) {
  return (
    <View className="bg-white rounded-lg p-4 shadow-md">
      <View className="flex-row items-center mb-4">
        <View className="w-16 h-16 bg-gray-200 rounded-full justify-center items-center mr-3">
          <Icon name="user-md" size={24} color="#4a6fa5" />
        </View>
        <View>
          <Text className="text-lg font-semibold text-gray-800">{doctor}</Text>
          <Text className="text-sm text-gray-600">{date}</Text>
        </View>
      </View>
      <ProfileField label="Motivo de consulta" value={reason} />
      <ProfileField label="Diagnóstico" value={diagnosis} />
      <ProfileField label="Notas médicas" value={notes} />
      <View className="mb-4">
        <Text className="text-sm font-semibold text-gray-800 mb-1">Receta médica</Text>
        <View className="flex-row justify-between items-center bg-gray-100 rounded-lg p-3">
          <View className="flex-row items-center">
            <Icon name="file-prescription" size={14} color="#6c757d" />
            <Text className="text-sm text-gray-600 ml-2">{prescription}</Text>
          </View>
          <TouchableOpacity className="border border-blue-600 rounded-lg px-2 py-1" onPress={onDownload}>
            <Icon name="download" size={14} color="#4a6fa5" />
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}