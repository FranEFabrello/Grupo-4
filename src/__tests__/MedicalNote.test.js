import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import MedicalNote from '~/components/MedicalNote'

// Mock dependencies
jest.mock('react-native-vector-icons/FontAwesome5', () => 'Icon');
jest.mock('~/components/ProfileField.jsx', () => {
  const React = require('react');
  const { View, Text } = require('react-native');
  return ({ label, value }) => (
    <View>
      {label && <Text>{label}</Text>}
      <Text>{value}</Text>
    </View>
  );
});
jest.mock('react-i18next', () => ({ useTranslation: () => ({ t: key => key }) }));

describe('MedicalNote', () => {
  const props = {
    doctor: 'Dr. Smith',
    date: '2025-07-10',
    reason: 'Checkup',
    diagnosis: 'Healthy',
    notes: 'All good',
    prescription: 'Vitamin C',
    onDownload: jest.fn(),
  };

  it('renders all fields correctly and matches snapshot', () => {
    const { toJSON, getByText } = render(<MedicalNote {...props} />);
    expect(getByText('Dr. Smith')).toBeTruthy();
    expect(getByText('2025-07-10')).toBeTruthy();
    expect(getByText('Checkup')).toBeTruthy();
    expect(getByText('Healthy')).toBeTruthy();
    expect(getByText('All good')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

});