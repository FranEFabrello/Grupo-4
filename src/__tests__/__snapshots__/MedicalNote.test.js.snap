// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MedicalNote renders all fields correctly and matches snapshot 1`] = `
<View
  className="bg-white rounded-lg p-4 shadow-md"
>
  <View
    className="flex-row items-center mb-4"
  >
    <View
      className="w-16 h-16 bg-gray-200 rounded-full justify-center items-center mr-3"
    >
      <Icon
        color="#4a6fa5"
        name="user-md"
        size={24}
      />
    </View>
    <View>
      <Text
        className="text-lg font-semibold text-gray-800"
      >
        Dr. Smith
      </Text>
      <Text
        className="text-sm text-gray-600"
      >
        2025-07-10
      </Text>
    </View>
  </View>
  <View>
    <Text>
      Motivo de consulta
    </Text>
    <Text>
      Checkup
    </Text>
  </View>
  <View>
    <Text>
      Diagnóstico
    </Text>
    <Text>
      Healthy
    </Text>
  </View>
  <View>
    <Text>
      Notas médicas
    </Text>
    <Text>
      All good
    </Text>
  </View>
  <View
    className="mb-4"
  >
    <Text
      className="text-sm font-semibold text-gray-800 mb-1"
    >
      medical_notes.prescription
    </Text>
    <View
      className="flex-row justify-between items-center bg-gray-100 rounded-lg p-3"
    >
      <View
        className="flex-row items-center"
      >
        <Icon
          color="#6c757d"
          name="file-prescription"
          size={14}
        />
        <Text
          className="text-sm text-gray-600 ml-2"
        >
          Vitamin C
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Icon
          color="#4a6fa5"
          name="download"
          size={14}
        />
      </View>
    </View>
  </View>
</View>
`;
